pkg/pubsub/node_data.go

```diff
@@ -220,10 +220,20 @@ func (n *NodeData) GetAccumulatedUptime() time.Duration {
 // Otherwise, it uses the time since the last joined event.
 func (n *NodeData) UpdateAccumulatedUptime() {
 	if n.Activity == ActivityLeft {
-		n.AccumulatedUptime += time.Since(time.Unix(n.LastLeftUnix, 0))
-		return
+		// Calculate the uptime for the most recent active period
+		recentUptime := n.LastLeftUnix - n.LastJoinedUnix
+		// Add this to the accumulated uptime
+		n.AccumulatedUptime += time.Duration(recentUptime) * time.Second
+	} else if n.Activity == ActivityJoined {
+		// If the node is currently active, calculate the uptime since it first joined
+		// This should only be done if the node is active and hasn't been updated yet
+		currentUptime := time.Now().Unix() - n.FirstJoinedUnix
```

Comment:
I'm not sure here, just for reference `n.FirstJoinedUnix` is updated only the first time a node is seen : https://github.com/masa-finance/masa-oracle/blob/3d09bb799e6689feb1e9bba1163174e11e409643/pkg/pubsub/node_data.go#L155 , but doesn't cover the case a node joins, stops, and then starts again, in that case the uptime would take account also of the shutdown time.

Reply:
```suggestion
		currentUptime := time.Now().Unix() - n.LastJoinedUnix
```